src/engagement_db_coda_sync/engagement_db_to_coda.py

```diff
@@ -172,20 +180,26 @@ def sync_engagement_db_to_coda(engagement_db, coda, coda_config, cache_path=None
         log.info(f"Initialising Coda sync cache at '{cache_path}/engagement_db_to_coda'")
         cache = CodaSyncCache(f"{cache_path}/engagement_db_to_coda")
 
+    if dry_run:
+        log.warning("Running without --dry-run may cause more reads than suggested here, because any update made to "
+                    "an engagement db message when syncing it will result in it being synced again")
+
     # Sync each dataset in turn to Coda
     dataset_to_sync_stats = dict()  # of engagement db dataset -> EngagementDBToCodaSyncStats
     for dataset_config in coda_config.dataset_configurations:
         log.info(f"Syncing engagement db dataset {dataset_config.engagement_db_dataset} to Coda dataset "
                  f"{dataset_config.coda_dataset_id}...")
-        dataset_sync_stats = _sync_engagement_db_dataset_to_coda(engagement_db, coda, coda_config, dataset_config, cache)
+        dataset_sync_stats = _sync_engagement_db_dataset_to_coda(engagement_db, coda, coda_config, dataset_config, cache, dry_run)
         dataset_to_sync_stats[dataset_config.engagement_db_dataset] = dataset_sync_stats
 
+    dry_run_text = "(dry run)" if dry_run else ""
+
     # Log the summaries of actions taken for each dataset then for all datasets combined.
     all_sync_stats = EngagementDBToCodaSyncStats()
     for dataset_config in coda_config.dataset_configurations:
         log.info(f"Summary of actions for engagement db dataset '{dataset_config.engagement_db_dataset}':")
         dataset_to_sync_stats[dataset_config.engagement_db_dataset].print_summary()
         all_sync_stats.add_stats(dataset_to_sync_stats[dataset_config.engagement_db_dataset])
 
-    log.info(f"Summary of actions for all datasets:")
+    log.info(f"Summary of actions for all datasets: {dry_run_text}")
```

Comment:
```suggestion
    log.info(f"Summary of actions for all datasets with dry run set to : {dry_run}")
```
Would be more expressive

Reply:
I think we should not change that because of consistency with other scripts i.e., restore_engagement_db in Tools repo and also it will be easier to figure if a run is in dry or not if you spot `dry run` in the logs instead of checking if it has been set to True or False.

Reply:
I still think printing the bool for a dry run gives more information, e.g when compared to printing empty string when the dry run is False. If you agree let's update restore_engagement_db to match the new log format in a diff PR. 

(am not strongly inclined to any of them - pick any that you think is more descriptive)

Reply:
I agree with @DanielMwendwa - if you haven't opted into dry-run then you shouldn't need to see it in the logs, and the expectation should be if the log says something has been done, then that's what was done unless it caveats that with the dry_run note.

(The compromise option would be to print (dry run) or something like (production run/non dry run) instead of True/False)